pkg/registrar/map_functions.go

```diff
@@ -40,6 +40,53 @@ type Mapper struct {
 	Logger Logger
 }
 
+func (mapper *Mapper) TemplateToWorkloadRequests(template client.Object) []reconcile.Request {
+	supplyChains := mapper.templateToSupplyChains(template)
+
+	var requests []reconcile.Request
+	for _, supplyChain := range supplyChains {
+		reqs := mapper.ClusterSupplyChainToWorkloadRequests(supplyChain)
+		requests = append(requests, reqs...)
+	}
+
+	return requests
+}
+
+func (mapper *Mapper) templateToSupplyChains(template client.Object) []*v1alpha1.ClusterSupplyChain {
+	templateName := template.GetName()
+
+	err := mapper.addGVK(template)
+	if err != nil {
+		mapper.Logger.Error(err, fmt.Sprintf("could not get GVK for template: %s", templateName))
+		return nil
+	}
+
+	list := &v1alpha1.ClusterSupplyChainList{}
+
+	err = mapper.Client.List(
+		context.TODO(),
+		list,
+	)
+
+	if err != nil {
+		mapper.Logger.Error(err, "list ClusterSupplyChains")
+		return nil
+	}
+
+	templateKind := template.GetObjectKind().GroupVersionKind().Kind
+
+	var supplyChains []*v1alpha1.ClusterSupplyChain
+	for scIndex := range list.Items {
+		sc := list.Items[scIndex]
```

Comment:
```suggestion
	for _, sc := range list.Items {
```
nitpick- this would be more consistent with what we do in `ClusterDeliveryToDeliverableRequests` and `ClusterSupplyChainToWorkloadRequests` below and is a bit easier to reason about while reading through the code.

Reply:
That actually causes major problems. See https://play.golang.org/p/GzU80LswSEt

Reply:
we also inlined a comment to guide people.